fhd_core/HEALPix/healpix_cnv_generate.pro

```diff
@@ -36,7 +36,9 @@ ENDIF ELSE radius=hpx_radius
 
 ;check if a string, if it is assume it is a filepath to a save file with the desired indices 
 ; (will NOT be over-written with the indices)
-IF Keyword_Set(restrict_hpx_inds) AND (size(restrict_hpx_inds,/type) NE 7) THEN restrict_hpx_inds=observation_healpix_inds_select(obs)
+IF Keyword_Set(restrict_hpx_inds) AND (size(restrict_hpx_inds,/type) NE 7) THEN BEGIN
+      IF restrict_hpx_inds NE 0 THEN  restrict_hpx_inds=observation_healpix_inds_select(obs)
+ENDIF
```

Comment:
This change should be removed. `Keyword_Set(restrict_hpx_inds)` already returns `False` if `restrict_hpx_inds=0`